packages/attribute-registry-process/src/services/attributeRegistryService.ts

```diff
@@ -1 +1,123 @@
-// TODO
+import {
+  AuthData,
+  CreateEvent,
+  eventRepository,
+  initDB,
+} from "pagopa-interop-commons";
+import {
+  AttributeEvent,
+  Attribute,
+  WithMetadata,
+  attributeEventToBinaryData,
+  attributeKind,
+} from "pagopa-interop-models";
+import { v4 as uuidv4 } from "uuid";
+import { config } from "../utilities/config.js";
+import {
+  ApiDeclaredAttributeSeed,
+  ApiVerifiedAttributeSeed,
+} from "../model/types.js";
+import { toCreateEventAttributeAdded } from "../model/domain/toEvent.js";
+import {
+  attributeDuplicate,
+  originNotCompliant,
+} from "../model/domain/errors.js";
+import { readModelService } from "./readModelService.js";
+
+const repository = eventRepository(
+  initDB({
+    username: config.eventStoreDbUsername,
+    password: config.eventStoreDbPassword,
+    host: config.eventStoreDbHost,
+    port: config.eventStoreDbPort,
+    database: config.eventStoreDbName,
+    schema: config.eventStoreDbSchema,
+    useSSL: config.eventStoreDbUseSSL,
+  }),
+  attributeEventToBinaryData
+);
+
+export const attributeRegistryService = {
+  async createDeclaredAttribute(
+    apiDeclaredAttributeSeed: ApiDeclaredAttributeSeed,
+    authData: AuthData
+  ): Promise<string> {
+    if (authData.externalId.origin !== "IPA") {
+      throw originNotCompliant("IPA");
+    }
+
+    return repository.createEvent(
+      createDeclaredAttributeLogic({
+        attribute: await readModelService.getAttributeByName(
+          apiDeclaredAttributeSeed.name
+        ),
+        apiDeclaredAttributeSeed,
+      })
+    );
+  },
+  async createVerifiedAttribute(
+    apiVerifiedAttributeSeed: ApiVerifiedAttributeSeed,
+    authData: AuthData
+  ): Promise<string> {
+    if (authData.externalId.origin !== "IPA") {
+      throw originNotCompliant("IPA");
+    }
+
+    return repository.createEvent(
+      createVerifiedAttributeLogic({
+        attribute: await readModelService.getAttributeByName(
+          apiVerifiedAttributeSeed.name
+        ),
+        apiVerifiedAttributeSeed,
+      })
+    );
+  },
+};
+
+export function createDeclaredAttributeLogic({
+  attribute,
+  apiDeclaredAttributeSeed,
+}: {
+  attribute: WithMetadata<Attribute> | undefined;
+  apiDeclaredAttributeSeed: ApiDeclaredAttributeSeed;
+}): CreateEvent<AttributeEvent> {
+  if (attribute) {
+    throw attributeDuplicate(apiDeclaredAttributeSeed.name);
+  }
+
+  const newDeclaredAttribute: Attribute = {
+    id: uuidv4(),
+    kind: attributeKind.declared,
+    name: apiDeclaredAttributeSeed.name,
+    description: apiDeclaredAttributeSeed.description,
+    creationTime: new Date(),
```

Comment:
As per [this comment](https://github.com/pagopa/interop-be-monorepo/pull/111#discussion_r1404274414) from @galales I suggest to use the following:

```
export function getCurrentTimeInUTC(): Date {
    return new Date((new Date()).toISOString());
}
```
What do you think @Carminepo2 ?
Should we include it in `commons` in a new `util.ts`?

Reply:
isn't this equivalent to `new Date()`?

btw more considerations about my previous comment: 
- the question was more like "when this Date object is stringified, are we sure that the output is the existing format?" 
- if we need to use a `Date` for business logic purposes, it's ok to leave a non-string version.
But if we need the string representation, we need to be sure that the format is defined and backward compatible (not sure if this definition can be centralized, in alternative a function in commons may be a viable solution)

Reply:
> isn't this equivalent to `new Date()`?
> 
> btw more considerations about my previous comment:
> 
>     * the question was more like "when this Date object is stringified, are we sure that the output is the existing format?"
> 
>     * if we need to use a `Date` for business logic purposes, it's ok to leave a non-string version.
>       But if we need the string representation, we need to be sure that the format is defined and backward compatible (not sure if this definition can be centralized, in alternative a function in commons may be a viable solution)

You're right, if we don't need the `Date` object then we can save the date as string directly using `.toISOString()` which saves the date in the format we need without the precision to milliseconds, but this isn't a problem as the standard is the same and can still be compared without problems